File: java-spring/src/main/java/test_service/v2/implementations/EchoService.java
package test_service.v2.implementations;

import org.springframework.stereotype.Service;
import test_service.v2.models.Message;
import test_service.v2.services.*;

@Service("EchoServiceV2")
public class EchoService implements IEchoService {

	@Override
	public EchoBodyResponse echoBody(Message body) {
		return new EchoBodyResponse(body);
	}
}

File: scala-play/app/v2/services/EchoService.scala
package v2.services

import javax.inject._
import scala.concurrent._
import v2.services.echo._
import v2.models._

@Singleton
class EchoService @Inject()()(implicit ec: ExecutionContext) extends IEchoService {
  override def echoBody(body: Message): Future[Message] = Future {
    Message(body.boolField, body.stringField)
  }
}
